src/Helpers.jl

```diff
@@ -44,3 +44,7 @@ function nextindex!(gen::IndexCounter)
     return letter(Threads.atomic_add!(gen.counter, 1))
 end
 resetindex!(gen::IndexCounter) = letter(Threads.atomic_xchg!(gen.counter, 1))
+
+# eps wrapper so it handles Complex numbers
+# if is Complex, extract the parametric type and get the eps of that
+wrap_eps(eltype) = eltype <: Real ? eps(eltype) : eps(first(eltype.parameters))
```

Comment:
this is not completely correct because we need to just fix it for `Complex` but there are `Real` types that are not `AbstractFloat` (`eps` is just defined for `AbstractFloat` types), like `Integer` or `Rational` for example. in those cases, it will fail with a cryptic message or might "fail successfully" with `Rational`.

I recommend you to implement it in this way:

```suggestion
wrap_eps(x) = eps(x)
wrap_eps(::Type{Complex{T}}) = eps(T)
```